internal/transport/grpcframer/framer.go

```diff
@@ -0,0 +1,267 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package grpcframer defines the interface for implementing an HTTP/2 framer,
+// its required types and an implementation of said framer.
+package grpcframer
+
+import "sync"
+
+// FrameType represents the type of a Frame and its value according to the
+// HTTP/2 spec.
+// See https://httpwg.org/specs/rfc7540.html#FrameTypes.
```

Comment:
Nit here and elsewhere in PR: no need to specify according to HTTP/2 spec in every comment. Perhaps "represents the type of HTTP/2 frame". No need to mention "value" either, that has no semantic meaning here in my opinion.

Reply:
I think there is value in linking to the appropriate portion of the HTTP/2 spec. Because the comment ends with a period, the URL is pointing to the top of the doc instead of to specific sections.

Reply:
Leaving the links in sgtm.